indica/readers/st40reader.py

```diff
@@ -390,7 +391,8 @@ def _get_charge_exchange(
                     self.QUANTITIES_MDS[instrument][q],
                     results["revision"],
                 )
-            except TreeNODATA:
+            except Exception as e:
```

Comment:
Generally it's better to catch specific exceptions rather than all of them. Should have a look at this when we rework the readers

Reply:
I completely agree!